src/test/java/teammates/storage/api/NotificationsDbTest.java

```diff
@@ -0,0 +1,229 @@
+package teammates.storage.api;
+
+import java.time.Instant;
+import java.util.UUID;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.NotificationType;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.storage.entity.Notification;
+import teammates.test.BaseTestCaseWithLocalDatabaseAccess;
+
+/**
+ * SUT: {@link NotificationsDb}.
+ */
+public class NotificationsDbTest extends BaseTestCaseWithLocalDatabaseAccess {
+
+    private final NotificationsDb notificationsDb = NotificationsDb.inst();
+
+    @Test
+    public void testGetNotification() throws Exception {
+        NotificationAttributes n = createNewNotification();
+
+        ______TS("typical success case");
+        NotificationAttributes retrieved = notificationsDb.getNotification(n.getNotificationId());
+        assertNotNull(retrieved);
+
+        ______TS("expect null for non-existent account");
+        retrieved = notificationsDb.getNotification("non.existent");
+        assertNull(retrieved);
+
+        ______TS("failure: null parameter");
+        assertThrows(AssertionError.class, () -> notificationsDb.getNotification(null));
+
+        // delete created account
+        notificationsDb.deleteNotification(n.getNotificationId());
+    }
+
+    @Test
+    public void testCreateNotification() throws Exception {
+        ______TS("typical success case");
+        String uuid = UUID.randomUUID().toString();
+        NotificationAttributes n = NotificationAttributes.builder(uuid)
+                .withTitle("title")
+                .withMessage("demo message")
+                .withType(NotificationType.DEPRECATION)
+                .withTargetUser(NotificationTargetUser.GENERAL)
+                .withStartTime(Instant.now())
+                .withEndTime(Instant.now().plusSeconds(3600))
+                .build();
+
+        notificationsDb.createEntity(n);
```

Comment:
Should we have an assert equals here here to check that it was created successfully or is it not needed because if an error was thrown the whole test would fail?
```
NotificationAttributes newNotif = notificationsDb.createEntity(n);
assertEquals(newNotif, n)
```

Reply:
Not much point to do that I think. Inside createEntity(), n is converted to entity and converted back to attributes. Value output should be the same by design. If anything wrong happens, it will have already thrown an Exception. This assert will not be even processed at that time.